### PR TITLE
Make changes for running analyticstack on 16.04.

### PR DESCRIPTION
**Pull request by brianhw**
* Replace llvm-gcc with gcc.
* Define new Hadoop services.
* Add retry when first using Hadoop.
* Add procps to debian packages to address bug.

Configuration Pull Request
---

Make sure that the following steps are done before merging

  - [ ] A devops team member has commented with :+1:

### CODE REVIEW

**Comment by brianhw** [comment]
This has finally been tested against a matching version of edx-analytics-pipeline (see https://github.com/edx/edx-analytics-pipeline/pull/379), and all acceptance tests pass.   (Or, rather, no acceptance tests fail.)

@edx/devops This is ready for final review.     FYI:  @tobz , @HassanJaveed84 , @azafty468.

@nedbat FYI. Once merged, ficus.master should be ready for ficus.3.

**Comment by brianhw** [comment]
@tobz I cloned your system definitions -- hopefully I did it the right way.

File: playbooks/roles/analytics_pipeline/tasks/main.yml [hunk 1]
```diff
@@ -127,21 +127,38 @@
...
-    name: hdfs
+    name: "{{ item }}"
+    enabled: yes
     state: started
+  with_items: "{{ hadoop_common_services }}"
   tags:
-    - manage
-    - manage:start
+    - install
+    - install:configuration
```
Comment (fredsmith) [inline playbooks/roles/analytics_pipeline/tasks/main.yml]: I think this is still manage, manage:start, not install, install:configuration.   Can you separate enabled and state into two tasks and tag the enabled:true one install, install:configuration and the state: started one manage, manage:start ?

File: playbooks/roles/hadoop_common/templates/hdfs-datanode.service.j2 [hunk 2]
```diff
@@ -0,0 +1,12 @@
+[Unit]
```
Comment (fredsmith) [inline playbooks/roles/hadoop_common/templates/hdfs-datanode.service.j2]: Can all these files go in the templates/etc/systemd/system/ directory in the role?  This is our standard practice for templates.

**Review by fredsmith — CHANGES_REQUESTED** [review]
Two minor chages, otherwise looks good.

File: playbooks/roles/analytics_pipeline/tasks/main.yml (hunk 1 above)
Comment (brianhw) [reply playbooks/roles/analytics_pipeline/tasks/main.yml]: I'll give it a try -- thanks!

File: playbooks/roles/hadoop_common/templates/hdfs-datanode.service.j2 (hunk 2 above)
Comment (brianhw) [reply playbooks/roles/hadoop_common/templates/hdfs-datanode.service.j2]: Yes, I will.

**Review by nedbat — APPROVED** [review]
Seems good, though I know nothing about hadoop-world.

**Comment by brianhw** [comment]
@fredsmith I made the changes you requested, rebuilt an analyticstack box, and ran some edx-analytics-pipeline acceptance tests against it successfully.  I ran some tests both before and after a "vagrant halt", to confirm that pipeline services were available.   Awaiting a thumb.

**Review by fredsmith — APPROVED** [review]

**Pull request merged**